inc/class-shortcode-ui.php

```diff
@@ -338,8 +324,66 @@ public function handle_ajax_do_shortcode() {
 		 */
 		do_action( 'shortcode_ui_after_do_shortcode', $shortcode );
 
-		wp_send_json_success( ob_get_clean() );
+		return ob_get_clean();
+	}
+
+	/**
+	 * Render a shortcode for preview in the TinyMCE editor.
+	 */
+	public function handle_ajax_do_shortcode() {
+
+		// Don't sanitize shortcodes — can contain HTML kses doesn't allow (e.g. sourcecode shortcode)
+		if ( ! empty( $_POST['shortcode'] ) ) {
+			$shortcode = stripslashes( $_POST['shortcode'] );
+		} else {
+			$shortcode = null;
+		}
+		if ( isset( $_POST['post_id'] ) ) {
+			$post_id = intval( $_POST['post_id'] );
+		} else {
+			$post_id = null;
+		}
+
+		if ( ! current_user_can( 'edit_post', $post_id ) || ! wp_verify_nonce( $_POST['nonce'], 'shortcode-ui-preview' ) ) {
+			echo esc_html__( "Something's rotten in the state of Denmark", 'shortcode-ui' );
+			exit;
+		}
 
+		wp_send_json_success( $this->render_shortcode_for_preview() );
 	}
 
+	/**
+	 * Get a bunch of shortcodes to render in preview.
+	 */
+	public function handle_ajax_bulk_do_shortcode() {
+
+		if ( is_array( $_POST['queries'] ) ) {
+
+			$responses = array();
+
+			foreach ( $_POST['queries'] as $posted_query ) {
+
+				// Don't sanitize shortcodes — can contain HTML kses doesn't allow (e.g. sourcecode shortcode)
+				if ( ! empty( $posted_query['shortcode'] ) ) {
+					$shortcode = stripslashes( $posted_query['shortcode'] );
+				} else {
+					$shortcode = null;
+				}
+				if ( isset( $posted_query['post_id'] ) ) {
+					$post_id = intval( $posted_query['post_id'] );
+				} else {
+					$post_id = null;
+				}
+
+				$responses[ $posted_query['counter'] ] = array(
+					'query' => $posted_query,
+					'response' => $this->render_shortcode_for_preview( $shortcode, $post_id ),
+				);
+			}
+
+			wp_send_json_success( $responses );
+			exit;
+		}
+
+	}
```

Comment:
I left the old method in there because I thought that it was handling something that core sent or expected. I guess not: removed in 87e7623.